app/controllers/income/estimatedPay/update/IncomeUpdatePayslipAmountController.scala

```diff
@@ -112,45 +122,55 @@ class IncomeUpdatePayslipAmountController @Inject()(
         {
-          val viewModel = {
-            val employer = IncomeSource(id = mandatorySeq(0).toInt, name = mandatorySeq(1))
-            val payPeriod = optionalSeq(0)
-            val payPeriodInDays = optionalSeq(1)
-            val taxablePayKey = optionalSeq(2)
-
-            val form = TaxablePayslipForm.createForm().fill(TaxablePayslipForm(taxablePayKey))
-            TaxablePaySlipAmountViewModel(form, payPeriod, payPeriodInDays, employer)
+          mandatorySeqEither match {
+            case Right(mandotorySeq) =>
+              val viewModel = {
+                val incomeSource = IncomeSource(id = mandotorySeq(0).toInt, name = mandotorySeq(1))
+                val payPeriod = optionalSeq(0)
+                val payPeriodInDays = optionalSeq(1)
+                val taxablePayKey = optionalSeq(2)
+
+                val form = TaxablePayslipForm.createForm().fill(TaxablePayslipForm(taxablePayKey))
+                TaxablePaySlipAmountViewModel(form, payPeriod, payPeriodInDays, incomeSource)
+              }
+              Ok(views.html.incomes.taxablePayslipAmount(viewModel))
+
+            case Left(_) => Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad())
+
           }
-          Ok(views.html.incomes.taxablePayslipAmount(viewModel))
         }
       }
   }
 
   def handleTaxablePayslipAmount: Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
     implicit val user = request.taiUser
 
-    val employerFuture = IncomeSource.create(journeyCacheService)
+    val incomeSourceFuture = IncomeSource.create(journeyCacheService)
 
     val futurePayPeriod = journeyCacheService.currentValue(UpdateIncome_PayPeriodKey)
     val futurePayPeriodInDays = journeyCacheService.currentValue(UpdateIncome_OtherInDaysKey)
     val futureTotalSalary = journeyCacheService.currentValue(UpdateIncome_TotalSalaryKey)
 
     (for {
-      employer        <- employerFuture
-      payPeriod       <- futurePayPeriod
-      payPeriodInDays <- futurePayPeriodInDays
-      totalSalary     <- futureTotalSalary
+      incomeSourceEither <- incomeSourceFuture
+      payPeriod          <- futurePayPeriod
+      payPeriodInDays    <- futurePayPeriodInDays
+      totalSalary        <- futureTotalSalary
     } yield {
       TaxablePayslipForm
         .createForm(FormHelper.stripNumber(totalSalary), payPeriod, payPeriodInDays)
         .bindFromRequest()
         .fold(
           formWithErrors => {
-            val viewModel = TaxablePaySlipAmountViewModel(formWithErrors, payPeriod, payPeriodInDays, employer)
-            Future.successful(BadRequest(views.html.incomes.taxablePayslipAmount(viewModel)))
+            incomeSourceEither match {
+              case Right(incomeSource) =>
+                val viewModel = TaxablePaySlipAmountViewModel(formWithErrors, payPeriod, payPeriodInDays, incomeSource)
+                Future.successful(BadRequest(views.html.incomes.taxablePayslipAmount(viewModel)))
```

Comment:
This will redirect you to the start of the journey if you enter incorrect values and mongo is down, not sure if we would prefer some content here instead, best to have a chat with a BA and UX

Reply:
Asked Andy, waiting for response

Reply:
This is the correct way to handle it.